GitUI/HelperDialogs/FormCommitDiff.cs

```diff
@@ -37,24 +38,40 @@ public FormCommitDiff(GitUICommands aCommands, string revisionGuid)
             }
         }
 
-        private void DiffFiles_SelectedIndexChanged(object sender, EventArgs e)
+        private async void DiffFiles_SelectedIndexChanged(object sender, EventArgs e)
         {
-            ViewSelectedDiff();
+            try
+            {
+                await ViewSelectedDiff();
+            }
+            catch (OperationCanceledException)
+            { }
         }
 
-        private void ViewSelectedDiff()
+        private async Task ViewSelectedDiff()
         {
             if (DiffFiles.SelectedItem != null && DiffFiles.Revision != null)
             {
                 Cursor.Current = Cursors.WaitCursor;
-                DiffText.ViewChanges(DiffFiles.SelectedItemParent?.Guid, DiffFiles.Revision?.Guid, DiffFiles.SelectedItem, String.Empty);
-                Cursor.Current = Cursors.Default;
+                try
+                {
+                    await DiffText.ViewChanges(DiffFiles.SelectedItemParent?.Guid, DiffFiles.Revision?.Guid, DiffFiles.SelectedItem, String.Empty);
+                }
+                finally
+                {
+                    Cursor.Current = Cursors.Default;
```

Comment:
I wonder if we should change cursor to the WaitCursor during an async operation.
I think we should not. The "awaiting" status should rather be indicated by the control awaiting the async call.

Reply:
I would make it waitcursor. The finally block will handle resetting it.

Reply:
Siding with Jay on this one

Reply:
Cursor is kind of global object if you have to different async actions in progress you are not able to properly use it to indicate you are awaiting an action. I find it weird to see wait cursor that is able to click on controls and making them act.

Reply:
Agree with Janusz.
"Only" about 50 places to change.

Reply:
So store an async count using interlocked to modify. If count > 0 leave cursor at wait. If count = 0 default.

Reply:
Actually changing the cursor does not work with async await, as soon as `await` is reached the cursor is changed accordingly to the pointed control. You can investigate it with
```csharp
Cursor.Current = Cursors.WaitCursor;
await Task.Delay(5000);
Cursor.Current = Cursors.Default;
```
The cursor changes to the wait cursor only for a fraction of second.

Reply:
So in the current implementation, the cursor is not changed. If async/await is the way forward, this will have to be changed to all callers set a temporary result. Cursor->message change can be done in this PR or in a separate PR taking a bigger grip.

However, what to do in release/2.51? Either the first patch that is similar to the 2.50 solution, skip updated cursor or rewrite callers there too?

Reply:
I would skip changing the cursor, the load time very short, I see no point in bothering this.

Reply:
I didn't give much though previously. I see a lot of sense in what Janusz and Gerhard are saying.

Reply:
```c#
            this.Cursor = Cursors.PanSW;
            await Task.Delay(5000);
            Cursor.Current = Cursors.Default;
            MessageBox.Show("Done");
```

Reply:
```c#
            Application.UseWaitCursor = true;
            await Task.Delay(5000);
            Application.UseWaitCursor = false;
            MessageBox.Show("Done");

Reply:
I propose that the Cursor is not changed and no changed message is setup.

I tried changing the text to "Waiting..." while getting the text. It is more annoying to see something that just flashes by that you do not have time to read here. Also it is hard to find every place in the forms where to set the text. For instance, the text should not be used when searching.
Also for longer operations (adding a delay), I see no benefit, you kind of know what is going on.

So I propose that Wait Cursor related to this is just removed, both for master and 2.51

Reply:
You don't get it. The delay was test code. Put it  in a form with a button. Make the button click async void. Put either sample in the click event.

Reply:
> You don't get it.

I get that you want the cursor. _I_ do not want the cursor changed here. I just have an opinion and leave it to @RussKie and @jbialobr to decide what to do.